src/datasources/balances-api/balances-api.manager.ts

```diff
@@ -0,0 +1,34 @@
+import { IConfigurationService } from '@/config/configuration.service.interface';
+import { IValkBalancesApi } from '@/datasources/balances-api/valk-balances-api.service';
+import { IBalancesApi } from '@/domain/interfaces/balances-api.interface';
+import { IBalancesApiManager } from '@/domain/interfaces/balances-api.manager.interface';
+import { Inject, Injectable } from '@nestjs/common';
+
+@Injectable()
+export class BalancesApiManager implements IBalancesApiManager {
+  private readonly valkBalancesChainIds: string[];
+  private readonly valkBalancesApi: IBalancesApi;
+
+  constructor(
+    @Inject(IConfigurationService)
+    private readonly configurationService: IConfigurationService,
+    @Inject(IValkBalancesApi) valkBalancesApi: IBalancesApi,
+  ) {
+    this.valkBalancesChainIds = this.configurationService.getOrThrow<string[]>(
+      'features.valkBalancesChainIds',
+    );
+    this.valkBalancesApi = valkBalancesApi;
+  }
+
+  isExternalized(chainId: string): boolean {
```

Comment:
Nit: I think we could improve the name a bit, e.g. something like `useProvider`, `useExternalApi`.

Reply:
I agree, changed to `useExternalApi` in [4f5723f](https://github.com/safe-global/safe-client-gateway/pull/1062/commits/4f5723f06f49a21b090547d02f90155e3a297cd2)